db/version_set.h

```diff
@@ -1265,6 +1278,7 @@ class VersionSet {
   WalSet wals_;
 
   std::unique_ptr<ColumnFamilySet> column_family_set_;
+  Cache* table_cache_;
```

Comment:
That's part of a small cleanup: before the patch, `VersionSet` did not have a pointer to the backing cache, so anytime it needed it, it had to retrieve it from `ColumnFamilySet`, resulting in awkward code like this for example:

```
void VersionSet::Reset() {
  if (column_family_set_) {
    Cache* table_cache = column_family_set_->get_table_cache();
    WriteBufferManager* wbm = column_family_set_->write_buffer_manager();
    WriteController* wc = column_family_set_->write_controller();
    column_family_set_.reset(
        new ColumnFamilySet(dbname_, db_options_, file_options_, table_cache,
                            wbm, wc, block_cache_tracer_, io_tracer_));
  }
```

Reply:
Maybe it's personal preference, and I may have been spoiled by powerful modern hardware. XD
Access to the table cache pointer is not frequent, thus I think it's fine to do the computation `Cache* table_cache = column_family_set_->get_table_cache();` each time. In such cases, I prefer doing computation on-demand (maybe using a function to wrap it) instead of introducing new data member to class.
Anyway, I do not have a strong opinion, and it's up to your choice.